pretty_midi/instrument.py

```diff
@@ -205,6 +210,163 @@ def get_end_time(self):
         else:
             return max(events)
 
+    def get_pitch_class_histogram(self, use_duration=False, use_velocity=False,
+                                  normalize=False):
+        """Computes the frequency of pitch classes of the current instrument,
+        optionally weighted by their durations or velocities.
+
+        Parameters
+        ----------
+        use_duration : bool
+            Weight frequency by note duration
+        use_velocity : bool
+            Weight frequency by note velocity
+        normalize : bool
+            Normalizes the histogram such that the sum of bin values is 1.
+
+        Returns
+        -------
+        histogram : np.ndarray, shape=(12,)
+            Histogram of pitch classes given current instrument, optionally
+            weighted by their durations or velocities
+        """
+
+        # Return all zeros if track is drum
+        if self.is_drum:
+            return np.zeros(12)
+
+        weights = np.ones(len(self.notes))
+
+        # Assumes that duration and velocity have equal weight
+        if use_duration:
+            weights *= [note.end - note.start for note in self.notes]
+        if use_velocity:
+            weights *= [note.velocity for note in self.notes]
+
+        histogram, _ = np.histogram([n.pitch % 12 for n in self.notes],
+                                    bins=np.arange(13),
+                                    weights=weights,
+                                    density=normalize)
+
+        return histogram
+
+    def get_pitch_class_transition_matrix(self, use_duration=False,
+                                          normalize=False):
+        """Computes the pitch class transition matrix of the current instrument,
+        optionally weighted by their durations.
+
+        Parameters
+        ----------
+        use_duration : bool
+            Increase frequency by transition duration (current and
+            next note)
+        normalize : bool
+            Normalize transition matrix such that matrix sum equals is 1.
+
+        Returns
+        -------
+        pitch_class_transition_matrix : np.ndarray, shape=(12,12)
+            Pitch class transition matrix
+        """
+
+        # track is drum or just one note, return all zeros transition matrix
+        if self.is_drum or len(self.notes) == 1:
+            return np.zeros((12, 12))
+
+        # container for holding pitch class transitions
+        pitch_class_transition_matrix = np.zeros((12, 12))
+
+        # self.notes is not sorted, sort them
+        sorted_notes = sorted(self.notes, key=lambda x: x.start)
+
+        # use 0.06s as the minimun time threshold for separate notes
+        time_thresh = 0.06
```

Comment:
I used 50 ms (period of 20 Hz) for the separation between onsets which denotes "rhythmic information".  Why did you choose 60 ms?